source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -24,7 +26,33 @@ void latchTime(Http::StreamDecoderFilterCallbacks* decoder_callbacks, absl::stri
   downstream_timing.setValue(key, decoder_callbacks->dispatcher().timeSource().monotonicTime());
 }
 
+// Helper function to apply filter state overrides to host and port.
+// Conditionally checks filter state based on the allow_dynamic_host_from_filter_state flag.
+auto applyFilterStateOverrides = [](absl::string_view& host, uint32_t& port,
```

Comment:
It is unconventional to use lambda in this way. Why not a function?

Reply:
Changed it to a function instead.